PublicMapping/districtbuilder
# Fix import region detection

## Overview

This fixes the region detection in the `ImportProjectScreen` component, and also adds an index to ensure uniqueness on `region_code` / `ccountry_code` for _visible_ regions, which should ensure that no more than 1 region config could ever be found by the new more comprehensive filter we're using on the frontend.

### Checklist


### Demo

![localhost_3003_create-project](https://user-images.githubusercontent.com/4432106/115618005-18e4cb00-a2c0-11eb-91cb-f8cdaaf8174e.png)


## Testing Instructions

- `scripts/migration`
- Ensure there is both a state `RegionConfig` that is not `hidden` and another `RegionConfig` with the same `region_code` that is `hidden` in your database (e.g. Wisconsin w/ `region_code` of WI and a hidden Dane County region with a `region_code` of WI, like on staging)
- Verify that you cannot insert a row into the `region_config` table with duplicate country / region code to an existing row unless it is `hidden`
- When exporting a WI map to CSV and importing it should detect the state as Wisconsin (not Dane County)
- When exporting a Dane County map to CSV and importing it should _also_ detect the state as Wisconsin (not Dane County)

Closes #683
